app/src/main/java/com/infomaniak/drive/data/services/UploadWorker.kt

```diff
@@ -400,7 +400,11 @@ class UploadWorker(appContext: Context, params: WorkerParameters) : CoroutineWor
                         processFoundLocalMedia(realm, cursor, contentUri, mediaFolder, syncSettings)
                     }
                 }
-        }.onFailure { exception -> syncMediaFolderFailure(exception, mediaFolder) }
+        }.onFailure { exception ->
+            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.Q && exception !is IllegalArgumentException) {
```

Comment:
You are throwing less exceptions than before.
Is it on purpose ?

The correct inverse of this `if` is :
`Build.VERSION.SDK_INT < Build.VERSION_CODES.Q || exception !is IllegalArgumentException`